nixos/modules/misc/nixpkgs.nix

```
@@ -112,6 +112,26 @@ in
       '';
     };
 
+    lib = mkOption {
+      default = lib;
+      defaultText = literalDocBook ''
+        the <literal>lib</literal> argument passed to modules
+      '';
+      example = literalExpression ''
+        lib.extend (final: prev: {
+          # Make lib.foo available in package expressions
+          foo = "bar";
+        })
+      '';
+      type = types.lazyAttrsOf types.unspecified;
```

Prohibit merging, as regular attrs merging doesn't compose with overlay style merging.

```suggestion
      type = types.unique
        { message = "`lib` is not suitable for plain attrset merging, as it does not interact well with the `extend` function. Any fixes to `lib` should be applied by invoking `extend` on some original `lib` value."; }
        (types.lazyAttrsOf types.raw);
```